File: app/src/main/java/com.example.petridelivery.wrappers/ClientWrapper.java
package com.example.petridelivery.wrappers;

import com.petri.delivery.web.objects.AdresaDto;
import com.petri.delivery.web.objects.ClientDto;
import com.petri.delivery.web.objects.NewClientBody;
import com.petri.delivery.web.objects.UpdateClientBody;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface ClientWrapper{

    String prefix = "/clienti/";

    @GET(prefix + "get_client")
    Call<List<ClientDto>> getClients(@Header("prenume") String prenume, @Header("nume") String nume,
                                    @Query("page") Integer page);

    @GET(prefix+"get_client_by_id")
    Call<ClientDto> getClient(@Query("id") Integer id);

    @GET(prefix+"get_clients")
    Call<List<ClientDto>> getClients(@Query("page") Integer page);

    @POST(prefix+"add_adresa")
    Call<Void> addAdresa(@Query("client_id") Integer client_id, @Body AdresaDto adresa);

    @DELETE(prefix+"remove_adresa")
    Call<Void> removeAdresa(@Query("client_id") Integer client_id, @Body AdresaDto adresa);

    @DELETE(prefix+"delete")
    Call<Void> deleteClient(@Query("id") Integer id);

    @POST(prefix+"add")
    Call<Void> addClient(@Body NewClientBody body);

    @POST(prefix+"update")
    Call<Void> updateClient(@Body UpdateClientBody body);
}

File: app/src/main/java/com.example.petridelivery.wrappers/ClientWrapper.kt
package com.example.petridelivery.wrappers

import com.petri.delivery.web.objects.AdresaDto
import com.petri.delivery.web.objects.ClientDto
import com.petri.delivery.web.objects.NewClientBody
import com.petri.delivery.web.objects.UpdateClientBody
import retrofit2.Call
import retrofit2.http.*

interface ClientWrapper {

    @GET("/clienti/get_client")
    suspend fun getClients(@Header("prenume") prenume: String?, @Header("nume") nume: String?,
                        @Query("page") page: Int?): MutableList<ClientDto>

    @GET("/clienti/get_client_by_id")
    suspend fun getClient(@Query("id") id: Int): ClientDto?

    @GET("/clienti/get_clients")
    suspend fun getClients(@Query("page") page: Int?): MutableList<ClientDto>

    @POST("/clienti/add_adresa")
    suspend fun addAdresa(@Query("client_id") client_id: Int, @Body adresa: AdresaDto)

    @DELETE("/clienti/remove_adresa")
    suspend fun removeAdresa(@Query("client_id") client_id: Int, @Body adresa: AdresaDto)

    @DELETE("/clienti/delete")
    fun deleteClient(@Query("id") id: Int): Call<Void>

    @POST("/clienti/add")
    suspend fun addClient(@Body body: NewClientBody)

    @POST("/clienti/update")
    suspend fun updateClient(@Body body: UpdateClientBody)
}